src/Mono.Android/Test/Xamarin.Android.Net/AndroidClientHandlerTests.cs

```diff
@@ -151,15 +151,37 @@ protected Task<HttpResponseMessage> ConnectIgnoreFailure (Func<Task<HttpResponse
 			try {
 				return connector ();
 			} catch (AggregateException ex) {
-				if (ex.InnerException is WebException wex && wex.Status == WebExceptionStatus.ConnectFailure) {
-					connectionFailed = true;
-					Assert.Ignore ($"Failed to connect to server. {wex}");
+				if (IgnoreIfConnectionFailed (ex.InnerException as WebException, out connectionFailed))
```

Comment:
Couldn't there be any number of inner exceptions that the `AggregateException` wraps?  Therefore, wouldn't it be prudent to examine *both* `AggregateException.InnerExceptions` *and* the `InnerException` chain?

```csharp
static IEnumerable<Exception> InnerExceptions (Exception e)
{
    while (e.InnerException != null) {
        yield return e.InnerException;
        e = e.InnerException;
    }
}
...
if (ex.InnerExceptions.SelectMany (ie => InnerExceptions (ie))
        .Any (ie => IgnoreIfConnectionFailed (ie as WebException, out connectionFailed))) { 
    return null;
}
```

Reply:
We don't want to catch any exception somewhere in the chain - we know that "our" exception we want to ignore is always the AggregateException's `InnerException`.